UI/cmake/feature-macos-update.cmake

```diff
@@ -0,0 +1,13 @@
+include_guard(DIRECTORY)
+
+if(NOT TARGET OBS::blake2)
+  add_subdirectory("${CMAKE_SOURCE_DIR}/deps/blake2" "${CMAKE_BINARY_DIR}/deps/blake2")
+endif()
+
+find_library(SECURITY Security)
+mark_as_advanced(SECURITY)
+
+target_sources(obs-studio PRIVATE update/crypto-helpers.hpp update/crypto-helpers-mac.mm update/shared-update.cpp
+                                  update/shared-update.hpp update/update-helpers.cpp update/update-helpers.hpp)
+
+target_link_libraries(obs-studio PRIVATE ${SECURITY} OBS::blake2)
```

Comment:
As this is an addition, use `"$<LINK_LIBRARY:FRAMEWORK,Security.framework>"` instead - it's the "correct" modern way to link system frameworks with CMake. It also makes the `find_library` call above unnecessary.